pyogrio/_ogr.pxd

```diff
@@ -14,7 +14,7 @@ cdef extern from "cpl_conv.h":
     void        CPLSetConfigOption(const char* key, const char* value)
 
 
-cdef extern from "cpl_error.h":
+cdef extern from "cpl_error.h" nogil:
```

Comment:
I haven't seen this usage.

I thought `nogil` was limited to either `with nogil:` or function declaration `cdef some_func() nogil:` blocks.

From the [Cython docs](https://cython.readthedocs.io/en/latest/src/userguide/external_C_code.html#including-verbatim-c-code), it looks like this would instead be added to individual C function declarations, like:

```
void CPLDefaultErrorHandler(CPLErr, int, const char *) nogil
```

Reply:
We actually already have it in one other place in this file as well, and it seems to work in practice. Without adding this, I get a cythonize error that I am calling a gil function in a nogil context:

```
Error compiling Cython file:
------------------------------------------------------------
...
    """
    if err_class == CE_Fatal:
        # If the error class is CE_Fatal, we want to have a message issued
        # because the CPL support code does an abort() before any exception
        # can be generated
        CPLDefaultErrorHandler(err_class, err_no, err_msg)
                             ^
------------------------------------------------------------

pyogrio/_err.pyx:224:30: Calling gil-requiring function not allowed without gil
```

I assume that it essentially is the equivalent of marking every entry in this block as nogil. If I only just mark `CPLDefaultErrorHandler` as nogil with:

```
cdef extern from "cpl_error.h":
    ...
    void CPLDefaultErrorHandler(CPLErr, int, const char *) nogil
    ...
```
then it works as well.

Reply:
It seems cython is also using this pattern in their own included pxd files, eg https://github.com/cython/cython/blob/c1025fcf9a7ef52c1fa81eceb7a49f257d8f0db8/Cython/Includes/libc/math.pxd#L1